git_workflow.md

```diff
@@ -0,0 +1,222 @@
+# Git and GitHub Workflow Guidelines
+
+Our workflow consists of 11 basic steps:
+
+- [ ] 1. [Set Configurations](#1)
+- [ ] 2. [Clone Repository](#2)
+3. [Create Issue](#3)
+4. [Create Branch](#4)
+5. [Develop Code](#5)
+6. [Commit Code](#6)
+7. [Merge Master](#7)
+8. [Open Pull Request](#8)
+9. [Pass Continuous Integration](#9)
+10. [Merge Development Into Master](#10)
+11. [Update Version](#11)
+
+Note: All code presented in this document are git commands from terminal. Many alternatives exist with Git GUIs or on GitHub.com.
+
+1.	Set Configurations <a name="1"/>
+
+For first time users:
+- Download necessary programs
+  * [Downloads](#README.md dls)
+
+- Set __global user options__ to identify your commits
+  * `git config --global user.name <name>`
+  * `git config --global user.email <email>`
+  * `git config --global core.editor <editor>` # e.g., vi, emacs, nano, TextWrangler (Microsoft Windows user refer to [First-Time-Git-Setup](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup))
+  * `git config --global merge.conflictstyle meld` # see merge section
```

Comment:
- I'm pretty sure that `meld` is not a `merge.conflictstyle`, see https://git-scm.com/docs/git-merge; `meld`, however, is one of the many options as `merge.tool` (it is not clear whether or not `meld` supports 3-way merge resolution: http://meldmerge.org/features.html -- they talk about it, but I don't see any example)
- I strongly argue that we require a 4-view (local, base, remote, merged) tool (e.g., https://stackoverflow.com/questions/161813/how-to-resolve-merge-conflicts-in-git) -- I personally still use the ancient `kdiff3` (http://kdiff3.sourceforge.net/doc/merging.html), but there are several other 4-way tools, but `meld` is not one of them.
- which requires that I argue strongly that we require a 3-way merge conflict resolution strategy, i.e., that `git config --global merge.conflictstyle diff3` (e.g., https://stackoverflow.com/questions/27417656/should-diff3-be-default-conflictstyle-on-git) -- too many of our programmers mess up the conflict resolution due to ignorance about ancestral state.